src/viam/examples/camera/example_camera.cpp

```diff
@@ -0,0 +1,106 @@
+#include <chrono>
+#include <cstddef>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <boost/optional.hpp>
+#include <boost/program_options.hpp>
+#include <grpcpp/channel.h>
+#include <grpcpp/client_context.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/support/status.h>
+#include <unistd.h>
+
+#include <viam/api/common/v1/common.pb.h>
+#include <viam/api/robot/v1/robot.grpc.pb.h>
+#include <viam/api/robot/v1/robot.pb.h>
+
+#include <viam/sdk/components/camera/camera.hpp>
+#include <viam/sdk/components/camera/client.hpp>
+#include <viam/sdk/components/camera/server.hpp>
+#include <viam/sdk/robot/client.hpp>
+#include <viam/sdk/robot/service.hpp>
+#include <viam/sdk/rpc/dial.hpp>
+
+using namespace viam::sdk;
+using namespace std;
+
+int main() {
+    // If you want to connect to a remote robot, this should be the url of the robot
+    // Ex: xxx.xxx.viam.cloud
+    string robot_address("localhost:8080");
+    // If you want to connect to a remote robot, you need the robot location secret (not
+    // the part secret) You can find this on app.viam.com
+    Credentials credentials("");
+
+    DialOptions dial_options;
+
+    // If you have credentials, use this to pass them to the robot
+    // dial_options.credentials = credentials;
+
+    // This is for an example. You should **not** have this option enabled in real systems.
+    dial_options.allow_insecure_downgrade = credentials.payload.length() == 0;
```

Comment:
Any thoughts on `= true` vs `= credentials.payload.length() == 0`? I did the latter in case someone accidentally copy&pastes this into prod without reading carefully.

Reply:
Changed to `.empty()` and moved discussion.